src/services/posts/fetch-posts.js

```diff
@@ -14,23 +14,54 @@ const fetchPosts = async(isFlagged) => {
 	 const fromClause = `FROM posts`
 	  
 	 const joinClause = `INNER JOIN users on posts."userId" = users.id`
-	 let whereClause = ''
+	 const whereClauseArray = []
 	 let orderByClause = `ORDER BY posts."createdAt" DESC`
-	   
-	 if(isFlagged !== undefined){
-		 selectClause+= `, posts."flagsCount"`
-		 
-		whereClause += `
-			WHERE posts."flagsCount" ${isFlagged ? '>': '='} 0`
+	 const limitClause = `LIMIT ${limit}`
+
+
+	// eslint-disable-next-line no-unused-expressions
+	cursor && whereClauseArray.push(` posts."id" > ${cursor}`)
+
+	if(isFlagged !== undefined){
+		selectClause+= `, posts."flagsCount"`
+		whereClauseArray.push(`posts."flagsCount"
+		 ${isFlagged ? '>': '='} 0`)
 		orderByClause =`ORDER BY "flagsCount" DESC`
-	 }
+	}
+	
+	const whereClause = (whereClauseArray.length > 0 ? ' WHERE ' : '') 
+	+ whereClauseArray.join(' AND ')
 
-	const feed = await db.query(`
-${selectClause} ${fromClause} ${joinClause} ${whereClause} ${orderByClause};
+	const result = await db.query(`
+${selectClause} ${fromClause} ${joinClause} 
+${whereClause} ${orderByClause} ${limitClause};
 `
 	) 
+	const feed = result.rows
+	return feed
+}
+
+const getTotalCount = async(isFlagged) => {
+	const selectClause = `SELECT COUNT(posts)`
+	const fromClause = `FROM posts`
+	let whereClause = ''
+
+	if(isFlagged !== undefined){
+	
+		whereClause = (`WHERE posts."flagsCount"
+		 ${isFlagged ? '>': '='} 0`)
+	}
+
+	const value = await db.query(` 
+	 ${selectClause} ${fromClause} ${whereClause};`)
+	const count = value.rows[0]
+	return Number(count.count)
+}
+const fetchPosts = async(isFlagged, cursor, limit = DEFAULT_LIMIT ) => {
 	
-	return feed.rows
+	const feed = await fetchData(isFlagged, cursor, limit) 
+	const count = await getTotalCount(isFlagged)
+	return {feed , count}
```

Comment:
this is cool. Now, as an improvement, we don't have to wait for data to be fetched before counting. 

We can perform both tasks together. Kindly look into [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all). Feel free to look at other resources in relation to it if necessary